transport/tchannel/config.go

```diff
@@ -32,11 +33,24 @@ import (
 
 const transportName = "tchannel"
 
+var defaultConnectionTimeout = 500 * time.Millisecond
+
 // TransportConfig configures a shared TChannel transport. This is shared
 // between all TChannel outbounds and inbounds of a Dispatcher.
 //
-// TransportConfig does not have any parameters at this time.
-type TransportConfig struct{}
+//  transports:
+//    tchannel:
+//      connection-timeout: 500ms
+type TransportConfig struct {
+	ConnectionTimeout time.Duration `config:"connection-timeout,interpolate"`
+}
+
+func newTransportConfig() transportConfig {
```

Comment:
This isn't actually how golang allocates structs that become function parameters - when structs are passed as function parameters, they actually are copied to a new location in the heap. Regardless, I wish we were more consistent with using just struct pointers.